li-hadoop-plugin/src/main/groovy/com/linkedin/gradle/libangbang/LiHadoopShellCommand.groovy

```diff
@@ -0,0 +1,28 @@
+/*
+ * Copyright 2015 LinkedIn Corp.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License"); you may not
+ * use this file except in compliance with the License. You may obtain a copy of
+ * the License at
+ *
+ * http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS, WITHOUT
+ * WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied. See the
+ * License for the specific language governing permissions and limitations under
+ * the License.
+ */
+
+package com.linkedin.gradle.libangbang;
+
+/**
+ * All the command for jobtypes which support bangbang should extend this class
+ */
+public abstract class LiHadoopShellCommand {
```

Comment:
Naming it LiBangBangCommand would mean that it extends BangBangCommand which is basically a command class for the bangbang type commands. LiHadoopShellCommand is actually just a wrapper over a HadoopShell command so It kind of makes sense. I have addressed your other comments but haven't changed this. Let me know if it doesn't work and I'll change it before merging.